5 feat 똑똑 공동탐색 페이지 구현 5

jyzzang02 opened the pull request:
### PR 타입(하나 이상의 PR 타입을 선택해주세요)
-기능 추가

### 반영 브랜치
5-feat-똑똑-공동탐색-페이지-구현-5-> develop

### 변경 사항
똑똑 공동탐색 기능의 xml 디자인 구현과 그에 필요한 아이콘 drawable을 추가하였습니다.
로직을 추가
NoiseInquiryActivity
NoiseInquiryViewModel
NoiseInquiryViewModelFactory
ResponseAdapter
step2.kt
step2Repository.kt

### 테스트 결과

KIMMJAE0 committed:
feat: step2 drawable 추가

step2 첫 페이지에 활용되는 drawable을 추가했습니다.

KIMMJAE0 committed:
design: step2 첫번째 화면 layout

똑똑 공동탐색의 첫 번째 화면 layout입니다.

jyzzang02 committed:
design: step2 실시간 응답현황 페이지

실시간 응답현황 페이지 디자인 입니다. 실제 응답 기반으로 하려면 추후에 수정이 필요할 수도 있겠습니다.

jyzzang02 committed:
design: step2 결과페이지

똑똑 공동탐색에서 결과 페이지입니다. drawable도 포함되어 있습니다.

jyzzang02 committed:
design: step2 전송 중 페이지

똑똑 공동탐색에서 알림을 전송하는 페이지입니다.

jyzzang02 committed:
design: step2 응답 받은 내역

똑똑 공동탐색을 했을때 다른 집 주인이 응답을 했을 경우activity_step2_responses.xml에 있는 recyclerview에 추가되게 됩니다.

jyzzang02 committed:
feat: step2 공동탐색 모델 및 리포지토리

똑똑 공동탐색의 모델 및 리포지토리를 구현해 봤습니다. 일단 임의의 값이나 확률을 넣어 두었습니다. 추후에 수정해야 할 듯합니다.

jyzzang02 committed:
feat: step2 로직 기본 설정

step2 똑똑 공동 탐색 로직 구현을 위한 기본 파일만 설정하였습니다. View, ViewModel, Factoy, Adapter

jyzzang02 committed:
fix

fix

jyzzang02 committed:
feat & fix: 리포지토리 일부 추가 및 drawable 수정

jyzzang02 committed:
feat: step2 viwModel 구현

HeeFlower5001 commented on app/src/main/java/com/kau/ttokttok/domain/model/step2.kt:
model 안에 step2 패키지를 따로 만들어서,
그 안에 하나의 파일에 하나의 클래스가 들어갈 수 있게끔 .kt 파일을 만들어서 분리시키는 게 좋을 것 같아요!
Enum class의 경우에는 model/step2/Enum 패키지를 하나 더 만들어서 분리를 하면 
나중에 수정을 하거나 관련 로직을 추가하고 확장할 때 관심사 분리를 할 수 있어요!

jyzzang02 replied:
말씀하신대로 수정하였습니다.

jyzzang02 committed:
feat: step2 viewModel Factory

ViewModel Factory를 구현하여 NoiseInquiryViewModel 생성 지원

안전한 타입의 ViewModel 생성 방식 적용

jyzzang02 committed:
design: ResponseAdapter에 활용할 것들 추가

필요한 drawable과 사용할 색깔을 추가했습니다.

jyzzang02 committed:
feat: view 및 Adapter 구현

jyzzang02 committed:
refactor: model 내부에 step2패키지 및 enum패키지 생성

말씀 하신대로 model 안에 step2 패키지를 만들고, enums도 따로 패키지로 만들었습니다.

jyzzang02 committed:
feat: Adapter 기본 구조 구현

- ListAdapter와 DiffUtil 기반 ResponseAdapter 생성
- ViewHolder에 기본 위치 텍스트 바인딩
- 효율적인 리스트 업데이트를 위한 DiffCallback 추가

jyzzang02 committed:
feat: Activity 1~4페이지 화면 전환 구조 구현

- ViewModel, Adapter 초기화 및 상태 관찰 로직
- 4단계 화면 전환을 위한 기본 구조
- ViewBinding 관리 및 메모리 누수 방지 처리

jyzzang02 committed:
feat: Adapter 응답 상태별 UI 변경 및 시간 표시 기능 추가

- 4가지 응답 타입별 아이콘, 색상, 텍스트 동적 변경
- 응답 시간 포맷팅 및 조건부 표시 로직
- HEARD/QUIET/SORRY/PENDING 상태별 시각적 구분

이로서 아답터는 일단 구현 완료

jyzzang02 committed:
feat: 카테고리 선택 페이지 및 버튼 로직

- 6가지 소음 카테고리 버튼 클릭 이벤트 연결
- 카테고리 선택 시 탐색 시작 로직

jyzzang02 committed:
feat: view 구현 완료

jyzzang02 commented:
<img width="352" height="756" alt="image" src="https://github.com/user-attachments/assets/b2aa6fe9-cf8d-4e1f-ac3a-e747451422bf" />
<img width="342" height="756" alt="image" src="https://github.com/user-attachments/assets/42a5d732-73f6-422d-83c8-e6dcf55eb636" />
<img width="336" height="760" alt="image" src="https://github.com/user-attachments/assets/43b1023e-dae1-4615-84d2-9e277efe0cc3" />
<img width="337" height="504" alt="image" src="https://github.com/user-attachments/assets/f532deac-511e-450b-adc2-7147f9165570" />
<img width="338" height="753" alt="image" src="https://github.com/user-attachments/assets/f329c27f-be73-4983-a6c4-1d926097b981" />
<img width="335" height="406" alt="image" src="https://github.com/user-attachments/assets/22071fdd-5575-4025-aa33-a2232400f45c" />
<img width="342" height="347" alt="image" src="https://github.com/user-attachments/assets/96a5614c-3efc-4d91-9402-65540c035356" />
